### PR TITLE
feat: :sparkles: mac address randomization

### LINKED ISSUE #72: Enable MAC Address randomization with a good opt-out
Pro: Mac addresses normally are device-unique and can completely de-anonymize you 

Con: randomize on every connect completely bloats the DHCP storage. I had problems using a repeater already, or people are randomly logged out, or speed problems, or crashes, ...

So this randomization should be opt-out but easily done, best in a setup dialog explaining how to exclude certain networks, and best already listing the available networks.

afaik poorly this is not integrated in KDE or GNOME yet, in an understandable way. GrapheneOS has this though

### PR DESCRIPTION
**Pull request by wandapeter**
Randomization tested and confirmed in VM
Fixes #72

### CODE REVIEW

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[device-mac-randomization]
+# "yes" is already the default for scanning
+wifi.scan-rand-mac-address=yes
+ 
+[connection-mac-randomization]
+# Randomize MAC for every ethernet connection
+ethernet.cloned-mac-address=random
+# Generate a random MAC for each WiFi and associate the two permanently.
+wifi.cloned-mac-address=stable
```
Comment (RoyalOughtness) [inline config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: This makes the mac address random per-network, not per-connection?

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[device-mac-randomization]
+# "yes" is already the default for scanning
+wifi.scan-rand-mac-address=yes
```
Comment (RoyalOughtness) [inline config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:3]: if this is already default, why do we need to set it?

**Comment by RoyalOughtness** [comment]
Please add evidence of the tests you successfully conducted

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf (hunk 2 above)
Comment (wandapeter) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:3]: i thought about that, then decided to leave it in bc i though declaring all randomization options explicitly is cleaner. Would you remove it?

Comment (RoyalOughtness) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:3]: No that's fine.

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf (hunk 1 above)
Comment (wandapeter) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: Maybe i misunderstood the discussion in #72. I thought the problem with random per connection is WIFI only. Should it be "per Network" on ethernet too?

Comment (RoyalOughtness) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: I'm just trying to understand what the config does. I'm not questioning it.

**Comment by RoyalOughtness** [comment]
Also, don't forget to add this to the list of hardening functionality in the readme :smile:

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf (hunk 1 above)
Comment (wandapeter) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: `stable` means that you will use the same MAC address every time you connect to that network.

**Comment by wandapeter** [comment]
> Please add evidence of the tests you successfully conducted

Tested in local build. The NetworkManager logs show setting a new MAC address on each connection and resetting it to the VM's default MAC on disconnect.

```bash
dev@fedora:~$ cat /usr/share/ublue-os/image-info.json 
{
"image-ref": "ostree-image-signed:docker://ghcr.io/ublue-os/silverblue-main-hardened",
"image-tag": "latest"
}
dev@fedora:~$journalctl -u NetworkManager.service | grep MAC
Nov 30 00:19:11 fedora NetworkManager[985]: <info>  [1701299951.8739] device (enp1s0): set-hw-addr: set-cloned MAC address to 8A:3D:57:4A:76:5E (random)
Nov 30 00:19:54 fedora NetworkManager[985]: <info>  [1701299994.2218] device (enp1s0): set-hw-addr: reset MAC address to 52:54:00:D4:63:02 (deactivate)
Nov 30 00:19:55 fedora NetworkManager[985]: <info>  [1701299995.2551] device (enp1s0): set-hw-addr: set-cloned MAC address to 0A:50:EC:9A:10:FE (random)
Nov 30 00:20:01 fedora NetworkManager[985]: <info>  [1701300001.0218] device (enp1s0): set-hw-addr: reset MAC address to 52:54:00:D4:63:02 (deactivate)
Nov 30 00:20:01 fedora NetworkManager[985]: <info>  [1701300001.5452] device (enp1s0): set-hw-addr: set-cloned MAC address to 6A:0B:56:2C:E2:96 (random)
Nov 30 00:52:56 fedora NetworkManager[985]: <info>  [1701301976.9189] device (enp1s0): set-hw-addr: reset MAC address to 52:54:00:D4:63:02 (deactivate)
Nov 30 00:54:02 fedora NetworkManager[985]: <info>  [1701302042.7801] device (enp1s0): set-hw-addr: set-cloned MAC address to DE:D7:46:69:86:5A (random)
```

File: config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf (hunk 1 above)
Comment (wandapeter) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: Thinking about it now `stable` randomization makes sense for wired connections too and having only one setting reduces the complexity too

Comment (RoyalOughtness) [reply config/files/usr/etc/NetworkManager/conf.d/rand_mac.conf:8]: Agreed.

**Review by RoyalOughtness — APPROVED** [review]

**Pull request merged**